framework/src/modules/chain/dpos/dpos.js

```diff
@@ -52,33 +60,46 @@ module.exports = class Dpos {
 		});
 	}
 
-	async getForgerPublicKeysForRound(round) {
-		return this.delegatesList.getForgerPublicKeysForRound(round);
+	async getForgerPublicKeysForRound(
+		round,
+		delegateListRoundOffset = this.delegateListRoundOffset,
+	) {
+		return this.delegatesList.getForgerPublicKeysForRound(
+			round,
+			delegateListRoundOffset,
+		);
 	}
 
 	async onBlockFinalized({ height }) {
 		this.finalizedBlockRound = this.slots.calcRound(height);
 	}
 
 	async onRoundFinish() {
-		// TODO use the configuration variable to set the value of this variable
-		const delegateListOffsetForRound = 2;
 		const disposableDelegateList =
-			this.finalizedBlockRound - delegateListOffsetForRound;
+			this.finalizedBlockRound - this.delegateListRoundOffset;
```

Comment:
same negative offset situation. Maybe we can just throw an error in the constructor?

Reply:
In this case the constructor is receiving this from config file which validates the value as grater than zero